test/com/google/enterprise/adaptor/fs/WindowsFileDelegateTest.java

```diff
@@ -306,63 +248,44 @@ public boolean GetFileSecurity(WString lpFileName,
-            String shareName, int level, PointerByReference bufptr) {
-          assertEquals(150, level);
-          bufptr.setValue(dfsInfo150);
-          return WinError.ERROR_SUCCESS;
-        }
-        @Override
-        public int NetApiBufferFree(Pointer buf) {
-          return WinError.ERROR_SUCCESS;
-        }
-      };
-
-    WindowsAclFileAttributeViews wafav =
-        new TestAclFileAttributeViews(null, null, null, netapi, null);
-    WindowsFileDelegate delegate =
-        new WindowsFileDelegate(advapi32, kernel32, netapi, wafav, 0);
-
-    return delegate.getDfsShareAclView(Paths.get("\\\\host\\namespace\\link"));
-  }
-
-  @Test
-  public void testGetDfsShareAclViewError() throws Exception {
-    Kernel32Ex kernel32 = new UnsupportedKernel32() {
-        @Override
-        public int GetLastError() {
-          // For when GetFileSecurity returns false.
-          return WinError.ERROR_ACCESS_DENIED;
-        }
-      };
-    Advapi32 advapi32 = new UnsupportedAdvapi32() {
-        @Override
-        public boolean GetFileSecurity(WString lpFileName,
-            int RequestedInformation, Pointer pointer, int nLength,
-            IntByReference lpnLengthNeeded) {
-          return false;
-        }
-      };
+    final Netapi32Ex.DFS_INFO_150 dfsInfo150 = new Netapi32Ex.DFS_INFO_150();
+    if (isLinkAcl) {
+      Memory daclMemory = new Memory(dacl.length);
+      daclMemory.write(0, dacl, 0, dacl.length);
+      dfsInfo150.SdLengthReserved = new ULONG(dacl.length);
+      dfsInfo150.pSecurityDescriptor = daclMemory;
+    } else {
+      dfsInfo150.SdLengthReserved = new ULONG(0);
+      dfsInfo150.pSecurityDescriptor = null;
+    }
+    dfsInfo150.write();
+    final Netapi32Ex.DFS_INFO_3 linkDfsInfo3 = newDfsInfo3(0x00000001);
+    final Netapi32Ex.DFS_INFO_3 namespaceDfsInfo3 = newDfsInfo3(0x00000101);
 
-    final Memory dfsInfo150 = new Memory(Native.LONG_SIZE + Pointer.SIZE);
-    dfsInfo150.setLong(0, 0L);
-    dfsInfo150.setPointer(Native.LONG_SIZE, null);
     Netapi32Ex netapi = new UnsupportedNetapi32() {
         @Override
         public int NetDfsGetInfo(String dfsPath, String serverName,
             String shareName, int level, PointerByReference bufptr) {
-          assertEquals(150, level);
-          bufptr.setValue(dfsInfo150);
-          return WinError.ERROR_SUCCESS;
+          if (level == 3) {
+            if (dfsPath.contains("link")) {
+              bufptr.setValue(linkDfsInfo3.getPointer());
+              return LMErr.NERR_Success;
+            } else if (dfsPath.contains("namespace")) {
+              bufptr.setValue(namespaceDfsInfo3.getPointer());
+              return LMErr.NERR_Success;
+            } else {
+              return WinError.ERROR_NOT_FOUND;
+            }
+          }
+          if (level == 150 && dfsPath.contains("link")) {
+            bufptr.setValue(dfsInfo150.getPointer());
+            return LMErr.NERR_Success;
```

Comment:
To match the actual doc for the calls, and because some of the code was taken from other mocks that did it right. They are the same value: 0.